code/lib/preview-api/src/modules/store/csf/portable-stories.ts

```diff
@@ -43,6 +89,12 @@ export function composeStory<TRenderer extends Renderer = Renderer, TArgs extend
     throw new Error('Expected a story but received undefined.');
   }
 
+  // users might pass an empty object instead of undefined e.g. composeStory(story, meta, {}, exportsName)
+  // and likely they expect the default project annotations to be used instead of completely resetting them
+  if (typeof projectAnnotations === 'object' && Object.keys(projectAnnotations).length === 0) {
+    projectAnnotations = GLOBAL_STORYBOOK_PROJECT_ANNOTATIONS as ProjectAnnotations<TRenderer>;
+  }
```

Comment:
Let's talk about this when you're available, I have thoughts, I don't like this special-case.

Brainduming here so I don't forget:

As a user I'd expect this argument to _merge_ with the default project annotations and not completely override them. But if that happens, how do you _disable/remove_ something from the default project annotations if you need to? I guess you can't? If you can't then we need to actually override them like we currently do. And if this API is a complete override, then as a user I wouldn't expect `{}` to work as `undefined`.
Vite configs have an elegant API for this issue but it might be too over-engineered in this case